packages/persist/lib/server.unit.test.ts

```diff
@@ -0,0 +1,197 @@
+                    trackDeletes: false
+                }),
+                headers: {
+                    'Content-Type': 'application/json'
+                }
+            });
+            expect(response.status).toEqual(201);
+        });
+    });
+
+    it('should delete records ', async () => {
+        const model = 'MyModel';
+        const records = [
+            { id: 1, name: 'r1' },
+            { id: 2, name: 'r2' }
+        ];
+        dbTracker.on('query', DBTracker.persistQueries(model));
+        const response = await fetch(`${serverUrl}/environment/123/connection/myconn/sync/abc/job/101/records`, {
+            method: 'DELETE',
+            body: JSON.stringify({
+                model,
+                records: records,
+                providerConfigKey: 'provider',
+                nangoConnectionId: 456,
+                activityLogId: 12,
+                lastSyncDate: new Date(),
+                trackDeletes: false,
+                softDelete: true
+            }),
+            headers: {
+                'Content-Type': 'application/json'
+            }
+        });
+        expect(response.status).toEqual(201);
+    });
+
+    it('should update records ', async () => {
+        const model = 'MyModel';
+        const records = [
+            { id: 1, name: 'r1' },
+            { id: 2, name: 'r2' }
+        ];
+        dbTracker.on('query', DBTracker.persistQueries(model));
+        const response = await fetch(`${serverUrl}/environment/123/connection/myconn/sync/abc/job/101/records`, {
+            method: 'PUT',
+            body: JSON.stringify({
+                model,
+                records: records,
+                providerConfigKey: 'provider',
+                nangoConnectionId: 456,
+                activityLogId: 12,
+                lastSyncDate: new Date(),
+                trackDeletes: false,
+                softDelete: true
+            }),
+            headers: {
+                'Content-Type': 'application/json'
+            }
+        });
+        expect(response.status).toEqual(201);
+    });
+});
+
+class DBTracker {
+    public static persistQueries(model: string) {
+        return (query: QueryDetails, step: number) => {
+            const steps = [
+                () => {
+                    query.response({ models: [model] } as SyncConfig);
+                },
+                () => {
+                    query.response([]);
+                },
+                () => {
+                    query.response([]);
+                },
+                () => {
+                    query.response([]);
+                },
+                () => {
+                    query.response([]);
+                },
+                () => {
+                    query.response({ id: 1 });
+                },
+                () => {
+                    query.response([]);
+                }
+            ];
+            steps[step - 1]?.();
+        };
+    }
+}
```

Comment:
ideally I would like to have a real in-memory database that can be used in tests. But knex in-memory solution is to use sqlite which doesn't support our schema